docs/reference/configuration.md

```diff
@@ -678,6 +679,102 @@ Defines the proto index of a field that can be used as the timestamp.
 * Example value: `2`
 * Type: `required (if SINK_PROM_WITH_EVENT_TIMESTAMP=true)`
 
+\`\`
+
+## MongoDB Sink
+
+A MongoDB sink Firehose \(`SINK_TYPE`= `mongodb` \) requires the following variables to be set along with Generic ones
+
+### `SINK_MONGO_CONNECTION_URLS`
+
+MongoDB connection URL/URLs to connect. Multiple URLs could be given in a comma separated format.
+
+* Example value: `localhost1:9200`
+* Type: `required`
+
+### `SINK_MONGO_DB_NAME`
+
+The name of the Mongo database to which you want to write the documents. If it does not exist, it will be created.
+
+* Example value: `sample_DB`
+* Type: `required`
+
+### `SINK_MONGO_AUTH_ENABLE`
+
+This field should be set to `true` if login authentication is enabled for the MongoDB Server.
+
+* Example value: `true`
+* Type: `optional`
+* Default value: `false`
+
+### `SINK_MONGO_AUTH_USERNAME`
+
+The login username for session authentication to the MongoDB Server. This is a required field is `SINK_MONGO_AUTH_ENABLE` is set to `true`
+
+* Example value: `bruce_wills`
+* Type: `optional`
+
+### `SINK_MONGO_AUTH_PASSWORD`
+
+The login password for session authentication to the MongoDB Server. This is a required field is `SINK_MONGO_AUTH_ENABLE` is set to `true`
+
+* Example value: `password@123`
+* Type: `optional`
+
+### `SINK_MONGO_AUTH_DB`
+
+The name of the Mongo authentication database in which the user credentials are stored. This is a required field is `SINK_MONGO_AUTH_ENABLE` is set to `true`
+
+* Example value: `sample_auth_DB`
+* Type: `optional`
+
+### `SINK_MONGO_COLLECTION_NAME`
+
+Defines the name of the Mongo Collection
+
+* Example value: `customerCollection`
+* Type: `required`
+
+### `SINK_MONGO_PRIMARY_KEY`
+
+The identifier field of the document in MongoDB. This should be the key of the field present in the message \(JSON or Protobuf\) and it has to be a unique, non-null field. So the value of this field in the message will be used as the ID of the document in MongoDB while writing the document.
+
+* Example value: `customer_id`
+* Type: `required`
```

Comment:
Hey Just wanted to know why is this a required field? If I understand correctly this would block anyone from using MongoDB's default `_id`(uuid) feature. Or is it a field other than `_id` in the document in mongo?

Reply:
suppose `MONGO_PRIMARY_KEY=driver_id` , so the value in the key `driver_id` , will be copied to the  `_id` field. so you basically overridding mongo's default `_id` . This is necessary to identify a document, so that update requests can be performed. Otherwise you'll have to create a new Index in Mongo to use another feild as a non-null, unique index.

Reply:
Exactly, shouldn't it be optional so that one could use `_id` and then create the separate index if and when needed? By this, aren't we restricting one not being able to use native mongo features? 🤔

Reply:
+1 to Akarsh. This should be optional.

Reply:
ok agreed, but we'll need another config parameter for the key which must be used to uniquely identify a document, in order to perform updates, and this must be a compulsory parameter
On a second thought, the default UUID provided by Mongo contains no useful info,so might not affect user so much

Reply:
because records arriving from Kafka will not know the UUID assigned by Mongo server, so need some other field to uniquely identify a document for update requests

Reply:
ok i get your point, working on it

Reply:
fixed